v3_async_wip/v3_async_wip/models.py

```diff
@@ -74,26 +86,72 @@ def set_as_security_message(self) -> None:
         """
         self._iothub_interface_id = constant.SECURITY_MESSAGE_INTERFACE_ID
 
-    def __str__(self) -> str:
-        return str(self.payload)
-
-    def get_size(self) -> int:
-        # TODO: this isn't actually accurate for what we use it for.
-        # Should we just remove it?
-        total = 0
-        total = total + sum(
-            sys.getsizeof(v)
-            for v in self.__dict__.values()
-            if v is not None and v is not self.custom_properties
-        )
-        if self.custom_properties:
-            total = total + sum(
-                sys.getsizeof(v) for v in self.custom_properties.values() if v is not None
-            )
-        return total
-
+    def get_system_properties_dict(self) -> Dict[str, str]:
+        """Return a dictionary of system properties"""
+        d = {}
+        # All messages
+        if self.message_id:
+            d["$.mid"] = self.message_id
+        if self.content_encoding:
+            d["$.ce"] = self.content_encoding
+        if self.content_type:
+            d["$.ct"] = self.content_type
+        # Outgoing Messages (D2C/Output)
+        if self.output_name:
+            d["$.on"] = self.output_name
+        if self._iothub_interface_id:
+            d["$.ifid"] = self._iothub_interface_id
+        # Incoming Messages (C2D/Input)
+        if self.input_name:
+            d["$.to"] = self.input_name
+        if self.ack:
+            d["iothub-ack"] = self.ack
+        if self.expiry_time_utc:
+            d["$.exp"] = self.expiry_time_utc
+        if self.user_id:
+            d["$.uid"] = self.user_id
+        if self.correlation_id:
+            d["$.cid"] = self.correlation_id
+        return d
 
-class MethodRequest(object):
+    @classmethod
+    # TODO: should this just replace the __init__?
```

Comment:
Thoughts?

Reply:
Yeah, maybe there's an advantage, but it feels like there are bigger fish to fry.